client/signup/controller.js

```diff
@@ -225,6 +226,31 @@ export default {
 
 		store.set( 'signup-locale', localeFromParams );
 
+		const isOnboardingFlow = flowName === 'onboarding';
+		if ( isOnboardingFlow ) {
+			const stepperOnboardingExperimentAssignment = await loadExperimentAssignment(
+				'calypso_signup_onboarding_stepper_flow'
+			);
+			if ( stepperOnboardingExperimentAssignment.variationName === 'stepper' ) {
+				const lang = localeFromParams ?? localeFromStore;
+				const step = getStepName( context.params ) ? `/${ getStepName( context.params ) }` : '';
+				const section = getStepSectionName( context.params )
+					? `/${ getStepSectionName( context.params ) }`
+					: '';
+				const locale = lang ? `/${ lang }` : '';
+
+				window.location.replace(
+					window.location.origin +
+						`/setup/onboarding` +
+						step +
+						section +
+						locale +
+						( context.querystring ? '?' + context.querystring : '' ) +
+						( context.hashstring ? '#' + context.hashstring : '' )
+				);
```

Comment:
I think this block could be simplified by using existing `getStepUrl`. We'd need to pass a `framework` param and update the internal one to either use that or extract locally. The result here could be a simpler call like:

```
window.location =
					getStepUrl(
						alternativeFlowName,
						getStepName( context.params ),
						getStepSectionName( context.params ),
						localeFromParams ?? localeFromStore
					) +
					( context.querystring ? '?' + context.querystring : '' ) +
					( context.hashstring ? '#' + context.hashstring : '' );
```

Reply:
I'm sure I wrote a comment about this somewhere, on how I thought about this but maybe it was overkill since it is temporary.

But since you're pointing to this too, and we may have more experiments of this type in the future, I'll convert to follow your suggestion.

Reply:
Cool. As you like. 👍